### PR TITLE
v3bwfile: stop using torflow_round_digs

### PR DESCRIPTION
**Pull request by juga0**
The generate cli argument changed name to round_digs but the
argument passed to V3BWFile.from_results was using the old
torflow_round_digs.

Closes bug #28602. Bugfix 1.0.3-dev0.

### CODE REVIEW

File: sbws/core/generate.py [hunk 1]
```diff
@@ -54,8 +54,13 @@ def gen_parser(sub):
     p.add_argument('-m', '--torflow-bw-margin', default=TORFLOW_BW_MARGIN,
                    type=float,
                    help="Cap maximum bw when scaling as Torflow. ")
-    p.add_argument('-r', '--round-digs', '--torflow-round-digs',
```
Comment (teor2345) [inline sbws/core/generate.py]: When --torflow-round-digs is passed to sbws, sbws should do what it did in 1.0.

Please revert your changes to lines 57 and 58.

File: sbws/core/generate.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 from sbws.globals import (fail_hard, SBWS_SCALE_CONSTANT, TORFLOW_SCALING,
                           SBWS_SCALING, TORFLOW_BW_MARGIN, PROP276_ROUND_DIG,
-                          DAY_SECS, NUM_MIN_RESULTS)
+                          DAY_SECS, NUM_MIN_RESULTS, TORFLOW_ROUND_DIG)
```
Comment (teor2345) [inline sbws/core/generate.py]: When you revert 57 and 58, you will also need to revert this change.

File: sbws/core/generate.py [hunk 3]
```diff
@@ -106,7 +111,7 @@ def main(args, conf):
     bw_file = V3BWFile.from_results(results, state_fpath, args.scale_constant,
                                     scaling_method,
                                     torflow_cap=args.torflow_bw_margin,
-                                    torflow_round_digs=args.torflow_round_digs,
+                                    round_digs=args.round_digs,
```
Comment (teor2345) [inline sbws/core/generate.py]: This change should work, because --round-digs is the first long option argument in the list:
https://docs.python.org/3/library/argparse.html#dest

Comment (teor2345) [reply sbws/core/generate.py]: This change should work without any changes to the arguments.

Comment (juga0) [reply sbws/core/generate.py]: Yes, but then you suggest but then we're never using --torflow-round-digs and we can't cause it won't be in the args Namespace.
You said in a previous comment:
> When --torflow-round-digs is passed to sbws, sbws should do what it did in 1.0.

Comment (teor2345) [reply sbws/core/generate.py]: I don't understand what the problem is here.

If you revert new lines 57-63 to old lines 57-58 in master, then:
1. args.round_digs should exist:
https://docs.python.org/3/library/argparse.html#dest
2. when --torflow-round-digs is passed to sbws, sbws will do what it did in 1.0

Can you show me a test where this does not work?

Comment (juga0) [reply sbws/core/generate.py]: > I don't understand what the problem is here.
> 
> If you revert new lines 57-63 to old lines 57-58 in master, then:
> 
>     1. args.round_digs should exist:
>        https://docs.python.org/3/library/argparse.html#dest
correct
>     2. when --torflow-round-digs is passed to sbws, sbws will do what it did in 1.0

no, --torflow-round-digs won't be in the namespace and it's not passed in V3BWFile.from_file in your suggestion.
> 
> Can you show me a test where this does not work?

I can create a test, but if you just checkout current master and run 'sbws generate`, you get:
`AttributeError: 'Namespace' object has no attribute 'torflow_round_digs'`
If you try `sbws generate --torflow-round-digs`, you get:
`sbws generate: error: argument -r/--round-digs/--torflow-round-digs: expected one argument`

Comment (juga0) [reply sbws/core/generate.py]: Oh, sorry, you were right, got confused here.
I couldn't revert just those lines because would lose other changes.
I created a fixup that reverts those lines, and use torflow's rounding method in the test.
And added a new test for those options.

**Comment by juga0** [comment]
squashed and merged.
Because  tests/unit/lib/test_v3bwfile.py was conflicting and was modified, GH didn't detect the branch merge.
